### PR TITLE
Enhancement/ts assertion

### PR DESCRIPTION
**Pull request by W3stside**
TS Friendly `assert`

depends on #190

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr219--dexreact.review.gnosisdev.com

File: src/utils/miscellaneous.ts [hunk 1]
```diff
@@ -1,4 +1,17 @@
 import { TokenDetails } from 'types'
+import { AssertionError } from 'assert'
+
+export function assert<T>(val: T, message: string): asserts val is NonNullable<T> {
```
Comment (Velenir) [inline src/utils/miscellaneous.ts]: Consider dropping this function when DefinitelyTyped/DefinitelyTyped#40415 is fixed

Comment (Velenir) [reply src/utils/miscellaneous.ts]: This actually asserts not only that val is NonNullable, but that it is truthy. No idea if it's possible to type that.

File: src/utils/miscellaneous.ts [hunk 2]
```diff
@@ -1,4 +1,17 @@
...
+  if (!val || val === undefined || val === null) {
```
Comment (Velenir) [inline src/utils/miscellaneous.ts]: `!val` is enough to check for `null and undefined`

File: src/api/exchange/DepositApiMock.ts [hunk 3]
```diff
@@ -94,7 +94,7 @@ export class DepositApiMock implements DepositApi {
 
     // Create the balance state if it's the first deposit
     const currentBatchId = await this.getCurrentBatchId()
-    let balanceState = this._initBalanceState(userAddress, tokenAddress, currentBatchId)
+    const balanceState = this._initBalanceState(userAddress, tokenAddress, currentBatchId)
```
[the remark this answers is not in the document]
Comment (W3stside) [reply src/api/exchange/DepositApiMock.ts]: think that was the linter that updated

**Pull request merged**